designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/CheckBox.kt

```diff
@@ -0,0 +1,103 @@
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+/**
+ * DHIS2 radio button with or without text. Wraps Material 3 [Checkbox].
+ *
+ * @param checked Controls if the option is checked.
+ * @param enabled Controls the enabled state of the button. When `false`, this button will not be
+ * clickable and will appear disabled to accessibility services.
+ * @param textInput The checkbox option text.
+ */
+@Composable
+fun CheckBox(
+    checked: MutableState<Boolean>,
+    enabled: Boolean,
+    textInput: String? = null
+) {
+    val interactionSource = remember { MutableInteractionSource() }
+
+    Row(
+        horizontalArrangement = Arrangement.spacedBy(0.dp, Alignment.Start),
+        verticalAlignment = Alignment.Top,
+        modifier = Modifier
+            .clickable(
+                interactionSource = interactionSource,
+                indication = null,
+                onClick = {
+                    if (enabled) {
+                        checked.value = !checked.value
+                    }
+                }
+            )
+    ) {
+        CompositionLocalProvider(LocalRippleTheme provides Ripple.CustomDHISRippleTheme) {
+            if (enabled) {
+                Checkbox(
+                    checked = checked.value,
+                    onCheckedChange = {
+                        checked.value = it
+                    },
+                    interactionSource = interactionSource,
+                    enabled = true,
+                    modifier = Modifier
+                        .size(Spacing.Spacing40),
+                    colors = CheckboxDefaults.colors(
+                        checkedColor = SurfaceColor.Primary,
+                        uncheckedColor = Outline.Dark,
+                        disabledCheckedColor = TextColor.OnDisabledSurface,
+                        disabledUncheckedColor = TextColor.OnDisabledSurface
+                    )
+                )
+            } else {
+                Checkbox(
+                    checked = checked.value,
+                    onCheckedChange = null,
+                    enabled = false,
+                    modifier = Modifier
+                        .size(Spacing.Spacing40),
+                    colors = CheckboxDefaults.colors(
+                        checkedColor = SurfaceColor.Primary,
+                        uncheckedColor = Outline.Dark,
+                        disabledCheckedColor = TextColor.OnDisabledSurface,
+                        disabledUncheckedColor = TextColor.OnDisabledSurface
+                    )
+                )
+            }
+        }
+        textInput?.let {
+            Text(
+                modifier = Modifier
+                    .padding(top = Spacing.Spacing8, bottom = Spacing.Spacing8),
+                text = it,
+                color = if (enabled) {
```

Comment:
could you shorthand this for better readability? i.e: if(enabled) TextColor.OnSurface else ..OnDisabled... or extract the color to a val.